utils/gear-replay-cli/README.md

```diff
@@ -4,174 +4,124 @@
 
 ```bash
-    gear-replay-cli -h
-    Commands of `gear-replay` CLI
+$ gear-replay-cli -h
+  Commands of `gear-replay` CLI
 
-    Usage: gear-replay-cli [OPTIONS] <COMMAND>
+  Usage: gear-replay-cli [OPTIONS] <COMMAND>
 
-    Commands:
-    replay-block  Replay block subcommand
-    gear-run      GearRun subcommand
-    help          Print this message or the help of the given subcommand(s)
+  Commands:
+    replay-block     Replay block subcommand
+    gear-run         GearRun subcommand
+    create-snapshot  Create a new snapshot file
+    help             Print this message or the help of the given subcommand(s)
 
-    Options:
-    -u, --uri <URI>            The RPC url [default: wss://archive-rpc.vara.network:443]
+  Options:
     -l, --log [<NODE_LOG>...]  Sets a custom logging filter. Syntax is `<target>=<level>`, e.g. -lsync=debug
     -h, --help                 Print help (see more with '--help')
 ```
 
 Currently supported cases include:
 
+- Preapring a snapshot file (which is extremely useful and allows to download large state once and replaying blocks multiple times)
 - Replaying a block on a live chain
+- Replaying a block on a snapshot
 
-  - current latest finalized block on Vara chain
+#### Creating state snapshot
+Create snapshot command:
 
-    ```bash
-    gear-replay-cli --uri wss://archive-rpc.vara.network:443 -lgear::runtime,pallet_gear,gear_common,pallet_gear_scheduler=debug replay-block
-    ```
+```bash
+$ gear-replay-cli create-snapshot -h
+  Create a new snapshot file
 
-  - block with `$HASH` or `$BLOCK_NUM`
+  Usage: gear-replay-cli create-snapshot [OPTIONS] [SNAPSHOT_PATH]
 
-    ```bash
-    export HASH=0x8dc1e32576c1ad4e28dc141769576efdbc19d0170d427b69edb2261cfc36e905
-    export BLOCK_NUM=2000000
+  Arguments:
+    [SNAPSHOT_PATH]  The snapshot path to write to
 
-    gear-replay-cli --uri wss://archive-rpc.vara.network:443 -lgear::runtime,pallet_gear,gear_common=debug replay-block --block "$HASH"
-    gear-replay-cli --uri wss://archive-rpc.vara.network:443 -lgear::runtime,pallet_gear,gear_common=debug replay-block --block "$BLOCK_NUM"
-    ```
+  Options:
+    -u, --uri <URI>                    The RPC url [default: wss://archive-rpc.vara.network:443]
+    -b, --block <BLOCK>                The block hash or number we want to replay. If omitted, the latest finalized block is used. The blockchain state at previous block with respect to this parameter will be scraped
+    -p, --pallet <PALLET>...           Pallet(s) to scrape. Comma-separated multiple items are also accepted. If empty, entire chain state will be scraped
+        --prefix <HASHED_PREFIXES>...  Storage entry key prefixes to scrape and inject into the test externalities. Pass as 0x prefixed hex strings. By default, all keys are scraped and included
+        --child-tree                   Fetch the child-keys as well
+    -h, --help                         Print help (see more with '--help')
+```
+Usage example:
+```bash
+export BLOCK_NUM=2000000
```

Comment:
If I'm not mistaken there is no mention about the env var in the help of create-snapshot subcommand. Would be nice to write a couple of words about that.

And am I right that the `-b`/`--block <BLOCK>` arguments have the priority over the environment variable?

Reply:
Maybe the doc is a bit misleading: the tool doesn't check any env vars - only command line arguments. It's just in the doc itself I use these BLOCK_NUM and HASH across a number of example commands so I figured they should be defined somewhere.
Apparently this causes ambiguity. Let me remove the env vars from the doc completely.